src/test/proxy/server_proxy.py

```diff
@@ -0,0 +1,138 @@
+from ja.user.config.add import AddCommandConfig
+from ja.user.message.query import QueryCommand
+from ja.user.message.cancel import CancelCommand
+from typing import List, Tuple
+import copy
+# from datetime import datetime
+
+
+class ServerProxyDummy(IUserServerProxy):
+
+    def __init__(self, ssh_config: SSHConfig):
+        self._ssh = ssh_config
+        self._jobs: List[Job] = []
+        self._counter = 0
+
+    @property
+    def jobs(self) -> List[Job]:
+        """
+        @return: the jobs on the server.
+        """
+        return self._jobs
+
+    def add_job(self, add_config: AddCommandConfig) -> Response:
+        job: Job = add_config.job
+        if job in self._jobs:
+            return Response(result_string="Job with id %s already exists" % job.uid,
+                            is_success=False)
+        job.uid = str(self._counter)
+        self._counter += 1
+        self._jobs.append(job)
+        job.status = JobStatus.QUEUED
+        return Response(result_string="Successfully added job with id: %s" % job.uid,
+                        is_success=True, uid=job.uid)
+
+    def cancel_job(self, cancel_command: CancelCommand) -> Response:
+        job_uid: str = cancel_command.uid
+        job_label: str = cancel_command.label
+        if job_uid is not None:
+            for job in self._jobs:
+                if job.uid == job_uid:
+                    job.status = JobStatus.CANCELLED
+                    return Response(result_string="Successfully removed job with id: %s" % job_uid,
+                                    is_success=True)
+            return Response(result_string="Job with id %s does not exist!" % job_uid,
+                            is_success=False)
+
+        if job_label is not None:
+            for job in self._jobs:
+                if job.label == job_label:
+                    job.status = JobStatus.CANCELLED
+                    print(job.status)
+                    return Response(result_string="Successfully removed job with label: %s" % job_label,
+                                    is_success=True)
+            return Response(result_string="Job with label %s does not exist!" % job_label,
+                            is_success=False)
+        return None
+
+    def query(self, query_command: QueryCommand) -> Response:
+        # Load parameters
+        uid: List[str] = query_command.uid
+        label: List[str] = query_command.label
+        owner: List[str] = query_command.owner
+        priority: List[JobPriority] = query_command.priority
+        status: List[JobStatus] = query_command.status
+        is_preemptible: bool = query_command.is_preemptible
+        special_resources: List[List[str]] = query_command.special_resources
+        cpu_threads: Tuple[int, int] = query_command.cpu_threads
+        memory: Tuple[int, int] = query_command.memory
+        # before: datetime = query_command.before
+        # after: datetime = query_command.after
+
```

Comment:
This will not be tested since the Jobs don't have an added property. I could however add it in the dummy somehow if that's really necessary.

Reply:
I think it's fine if you don't test for the time